### PR TITLE
Luis/rabbitmq streaming

### PR DESCRIPTION
**Pull request by Luishfs**
# Summary
This PR includes rabbitmq ( Pika) support for streaming pipelines
Users will be able to use a data loader block containing RabbitMQ, and consume messages and channel methods on transformer blocks

# Tests
Tested init_client on tests folder

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/executors/streaming_pipeline_executor.py [hunk 1]
```diff
@@ -99,10 +99,11 @@ def __execute_in_python(self):
         )
         sink = SinkFactory.get_sink(sink_config)
 
-        def handle_batch_events(messages: List[Union[Dict, str]]):
+        def handle_batch_events(messages: List[Union[Dict, str]],**kwargs):
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/executors/streaming_pipeline_executor.py:102]: nit: add a space before `**kwargs`

File: mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ const getDataSourceTypes = (
     return {
       [BlockTypeEnum.DATA_LOADER]: [
         DataSourceTypeEnum.KAFKA,
+        DataSourceTypeEnum.RABBITMQ,
```
Comment (wangxiaoyou1993) [inline mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.ts:28]: could you help sort this list in alphabetical order?

File: mage_ai/frontend/interfaces/DataSourceType.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ export enum DataSourceTypeEnum {
   GENERIC = 'generic',
   GOOGLE_CLOUD_STORAGE = 'google_cloud_storage',
   KAFKA = 'kafka',
+  RABBITMQ = 'rabbitmq',
```
Comment (wangxiaoyou1993) [inline mage_ai/frontend/interfaces/DataSourceType.ts:12]: nit: same. could you sort the names in alphabetical order? same in other places.

File: mage_ai/streaming/sources/rabbitmq.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+from collections import namedtuple
+import pika
+from pika.exceptions import AMQPConnectionError
+
+@dataclass
+class ConsumeConfig:
+    auto_ack: bool = False
+    exclusive: bool = False
+    inactivity_timeout: float = None
+
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/rabbitmq.py:14]: nit: remove extra empty lines. keep two lines between classes. could you use flake8 linter to check the code style?

File: mage_ai/streaming/sources/rabbitmq.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
+from mage_ai.shared.config import BaseConfig
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/rabbitmq.py:1]: nit: same here. sort the library in alphabetical order.

File: mage_ai/tests/streaming/sources/test_rabbitmq.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+                configure_consume= False,
+                username='guest',
+                password='guest',
+                amqp_url_virtual_host= r'%2f',
+                ))
+            self.assertTrue(
+                '__init__() missing 1 required positional argument: \'connection_host\''
+                in str(context.exception),
+            )
+            self.assertEqual(mock_init_client.call_count, 0)
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/streaming/sources/test_rabbitmq.py:35]: nit: add an empty line at the end of file.

File: mage_ai/streaming/sources/rabbitmq.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            self.create_connection = pika.BlockingConnection(pika.URLParameters(f"amqp://guest:guest@{self.config.connection_host}:{self.config.connection_port}/%2f"))
+            
+            self._print('Connected on broker')
+        
+
+            self.channel = self.create_connection.channel()
+
+
+        except AMQPConnectionError:
+            print('AMQP Connection Error, please check your broker connection or configuration')
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/rabbitmq.py:61]: nit: let's use `self._print` method

File: mage_ai/streaming/sources/rabbitmq.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+@dataclass
+class PikaConfig(BaseConfig):
+    connection_host: str
+    connection_port: int
+    queue_name: str
+    configure_consume: bool = False
+    username: str = 'guest'
+    password: str = 'guest'
+    amqp_url_virtual_host: str = r'%2f'
+    consume_config = ConsumeConfig = None
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/rabbitmq.py:28]: nit: `consume_config: ConsumeConfig = None`

File: mage_ai/streaming/sources/rabbitmq.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    def parse_config(self, config: Dict) -> Dict:
+        consume_config = config.get('consume_config')
+        if consume_config is not None and type(consume_config) is dict:
+            config['consume_config'] = ConsumeConfig(**consume_config)
+        return config
+
+    
+
+
+class PikaSource(BaseSource):
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/rabbitmq.py:40]: I would just call it RabbitMQSource and RabbitMQConfig.

File: mage_ai/frontend/interfaces/DataSourceType.ts (hunk 3 above)
Comment (Luishfs) [reply mage_ai/frontend/interfaces/DataSourceType.ts:12]: Noticed DATA_SOURCE_TYPES in same file are not in alpheticall order,  will be sorting them too

Comment (Luishfs) [reply mage_ai/frontend/interfaces/DataSourceType.ts:12]: ( respecting generic and source files on top)

File: mage_ai/data_preparation/executors/streaming_pipeline_executor.py (hunk 1 above)
Comment (Luishfs) [reply mage_ai/data_preparation/executors/streaming_pipeline_executor.py:102]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.ts (hunk 2 above)
Comment (Luishfs) [reply mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.ts]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/frontend/interfaces/DataSourceType.ts (hunk 3 above)
Comment (Luishfs) [reply mage_ai/frontend/interfaces/DataSourceType.ts]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/streaming/sources/rabbitmq.py (hunk 4 above)
Comment (Luishfs) [reply mage_ai/streaming/sources/rabbitmq.py:16]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/streaming/sources/rabbitmq.py (hunk 5 above)
Comment (Luishfs) [reply mage_ai/streaming/sources/rabbitmq.py]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/tests/streaming/sources/test_rabbitmq.py (hunk 6 above)
Comment (Luishfs) [reply mage_ai/tests/streaming/sources/test_rabbitmq.py]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/streaming/sources/rabbitmq.py (hunk 7 above)
Comment (Luishfs) [reply mage_ai/streaming/sources/rabbitmq.py]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/streaming/sources/rabbitmq.py (hunk 8 above)
Comment (Luishfs) [reply mage_ai/streaming/sources/rabbitmq.py]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

File: mage_ai/streaming/sources/rabbitmq.py (hunk 9 above)
Comment (Luishfs) [reply mage_ai/streaming/sources/rabbitmq.py]: Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**